modules/core/src/main/scala/gem/config/DynamicConfig.scala

```diff
@@ -14,29 +14,46 @@ sealed abstract class DynamicConfig extends Product with Serializable {
 
   def smartGcalKey: Option[SmartGcalKey] =
     this match {
-      case f2: F2Config     => Some(f2.key)
-      case _ => None
+      case f2: F2DynamicConfig => Some(f2.key)
+      case _                   => None
     }
 
 }
 object DynamicConfig {
   type Aux[I0] = DynamicConfig { type I = I0 }
+
+  sealed abstract class Impl[I0 <: Instrument with Singleton](val instrument: I0) extends DynamicConfig {
+    type I = I0
```

Comment:
Yes, I think it is path dependent.  I asked about`Singleton` too.  It's hard to find any documentation about.   [This entry](http://www.scala-archive.org/scala-Singleton-td1940630.html) is the most I could find.  Here I'm just aping what Rob did with the `StaticConfig` and creating instances for all the instruments so that the code compiles without warnings.  The goal though is to allow us to create an observation and know that its static and dynamic configs are consistent.   From #49:

```scala
type RealizedObservation =
  Observation[StaticConfig.Aux[A], Step[DynamicConfig.Aux[A]]] forSome { type A }
```